zmq.cpp

```diff
@@ -37,75 +39,95 @@ void LoadBalancer::run() {
         }
         // handle worker
         if (items[0].revents & ZMQ_POLLIN) {
+
             // the first frame is the identifier of the worker: we add it to the available worker
             avalailable_worker.push(z_recv(workers));
-            {
-                //  Second frame is empty
-                std::string empty = z_recv(workers);
-                assert(empty.size() == 0);
+
+            // the second frame should be empty
+            std::string empty = z_recv(workers);
+            if (empty != "") {
+                // just skip the rest of the message
+                continue;
             }
 
-            //  Third frame is READY or else a client reply address
-            std::string client_addr = z_recv(workers);
-
-            //  If client reply, send resp back to the appropriate client
-            if (client_addr != "READY") {
-                {
-                    // another empty frame
-                    std::string empty = z_recv(workers);
-                    assert(empty.size() == 0);
-                }
-                // the actual reply
-                zmq::message_t reply;
-                workers.recv(&reply);
-                z_send(clients, client_addr, ZMQ_SNDMORE);
-                z_send(clients, "", ZMQ_SNDMORE);
-                z_send(clients, reply);
+            frames.clear();
+            size_t more = 0;
+            size_t more_size = sizeof (more);
+            do {
+                std::string frame = z_recv(workers);
+                frames.push_back(frame);
+                // Are there more frames coming?
+                workers.getsockopt(ZMQ_RCVMORE, &more, &more_size);
+            } while (more);
+
+
+            if (frames.size() < 3) {
```

Comment:
Is there a reason for this condition? I wouldn't mind a comment to explain (probably because I don't know the underlying protocol).